packages/core-data/src/types/user.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Internal dependencies
+ */
+import { AvatarUrls, Context, ContextualField, OmitNevers } from './helpers';
+
+import { BaseEntityTypes as _BaseEntityTypes } from './base-entity-types';
+
+declare module './base-entity-types' {
+	export namespace BaseEntityTypes {
+		export interface User< C extends Context > {
+			/**
+			 * Unique identifier for the user.
+			 */
+			id: number;
+			/**
+			 * Login name for the user.
+			 */
+			username: ContextualField< string, 'edit', C >;
+			/**
+			 * Display name for the user.
+			 */
+			name: string;
+			/**
+			 * First name for the user.
+			 */
+			first_name: ContextualField< string, 'edit', C >;
+			/**
+			 * Last name for the user.
+			 */
+			last_name: ContextualField< string, 'edit', C >;
+			/**
+			 * The email address for the user.
+			 */
+			email: ContextualField< string, 'edit', C >;
+			/**
+			 * URL of the user.
+			 */
+			url: string;
+			/**
+			 * Description of the user.
+			 */
+			description: string;
+			/**
+			 * Author URL of the user.
+			 */
+			link: string;
+			/**
+			 * Locale for the user.
+			 */
+			locale: ContextualField< string, 'edit', C >;
+			/**
+			 * The nickname for the user.
+			 */
+			nickname: ContextualField< string, 'edit', C >;
+			/**
+			 * An alphanumeric identifier for the user.
+			 */
+			slug: string;
+			/**
+			 * Registration date for the user.
+			 */
+			registered_date: ContextualField< string, 'edit', C >;
+			/**
+			 * Roles assigned to the user.
+			 */
+			roles: ContextualField< string[], 'edit', C >;
+			/**
+			 * Password for the user.
+			 *
+			 * This is never sent from the server to the client
+			 * but exists because we might send an update to the
+			 * server with a new password to set.
+			 */
+			password?: string;
+			/**
+			 * All capabilities assigned to the user.
+			 */
+			capabilities: ContextualField<
+				Record< string, string >,
+				'edit',
+				C
+			>;
+			/**
+			 * Any extra capabilities assigned to the user.
+			 */
+			extra_capabilities: ContextualField<
+				Record< string, string >,
+				'edit',
+				C
+			>;
```

Comment:
These capabilities are actually `Record< string, boolean >` instead of `Record< string, string >` and these come from [here](https://github.com/WordPress/wordpress-develop/blob/b6f639737c5fa722dd5a80f0ad94e48c67c37ffd/src/wp-includes/class-wp-role.php#L25-L32).

Reply:
Created #68045 to fix it.